src/proc.c

```diff
@@ -290,6 +290,26 @@ int adjust_threads(int *tids, int nr_tids, int *user_tids,
     return 0;
 }
 
+void filter_threads(int *tids, int *index, char *states, int *nr_tids,
```

Comment:
Take `nr_tids` by value and return new count as a result?

Reply:
And what about using array syntax for arguments that are ... well ... arrays? :)

Reply:
Fixed